src/pathwayvis/components/map/map.component.ts

```diff
@@ -330,16 +327,27 @@ class MapComponentCtrl {
   /**
    * Callback function for clicked action button in context menu
    */
-  public processActionClick(action, data) {
-
+  public processActionClick(action, data, element) {
     if (action.type === 'reaction:link') {
       this.$window.open(`http://bigg.ucsd.edu/universal/reactions/${data.bigg_id}`);
     } else {
-      this._mapOptions.actionHandler(action, { id: data.bigg_id }).then((response) => {
-        this._mapOptions.setCurrentGrowthRate(parseFloat(response['growth-rate']));
-        this._mapOptions.setReactionData(response.fluxes);
-        this._mapOptions.setRemovedReactions(response['removed-reactions']);
-      });
+      this.shared.async(this._mapOptions.actionHandler(action, { id: data.bigg_id }).then((response) => {
+        this._mapOptions.updateMapData(response);
+        if (action.type.startsWith('reaction:objective')) {
+          if (action.type.endsWith('undo')) {
```

Comment:
We have a little bit too much of nested if statements here.This one could be simplified:

    this._mapOptions.setObjectiveReaction(action.type.endsWith('do') ? data.bigg_id : null)